sdk/src/iota-names-client.ts

```diff
@@ -249,6 +253,131 @@ export class IotaNamesClient {
+                        }
+                    }
+                }
+            `),
+            variables: {
+                address: iotaNamesObjectId,
+                type: getRegistryKeyType(packageId, getDenyListType(packageId)),
+                bcs: denyListBcsB64,
+            },
+        });
+
+        const denyList = response?.data?.owner?.dynamicField?.value?.json;
+        return {
+            reservedTableId: denyList?.reserved?.id ?? null,
+            blockedTableId: denyList?.blocked?.id ?? null,
+        };
+    }
+
+    /**
+     * Gets the full reserved/blocked list
+     */
+    async getRestrictedList(type: 'reserved' | 'blocked'): Promise<string[]> {
+        const { reservedTableId, blockedTableId } = await this.getDenyListTableIds();
+        if (type === 'reserved' && !reservedTableId) return [];
+        if (type === 'blocked' && !blockedTableId) return [];
+
+        const tableId = type === 'reserved' ? reservedTableId : blockedTableId;
+
+        const results: string[] = [];
+        let cursor: string | null = null;
+        const pageSize = 50;
+        let hasMorePages = true;
+
+        while (hasMorePages) {
+            const response: any = await this.graphQlClient.query({
+                query: graphql(`
+                    query getReservedListFromTable(
+                        $address: IotaAddress!
+                        $first: Int!
+                        $after: String
+                    ) {
+                        owner(address: $address) {
+                            dynamicFields(first: $first, after: $after) {
+                                pageInfo {
+                                    hasNextPage
+                                    endCursor
+                                }
+                                nodes {
+                                    name {
+                                        json
+                                    }
+                                }
+                            }
+                        }
+                    }
+                `),
+                variables: { address: tableId, first: pageSize, after: cursor },
+            });
+
+            const df = response?.data?.owner?.dynamicFields;
+            const nodes = df?.nodes ?? [];
+
+            for (const n of nodes) {
+                const json = n?.name?.json;
+
+                if (typeof json === 'string' && json.length > 0) {
+                    results.push(json);
+                    continue;
+                }
+
+                const bytes: number[] | undefined = json?.fields?.bytes;
+                if (Array.isArray(bytes) && bytes.length > 0) {
+                    try {
+                        const decoded = new TextDecoder().decode(new Uint8Array(bytes));
+                        if (decoded) results.push(decoded);
+                    } catch {
+                        // ignore undecodable entries
+                    }
+                }
+            }
```

Comment:
Why is it that this one needs more work to deserialize? I mean the whole bytes decoding

Maybe the graphql query is missing some param?